llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -2911,6 +2914,17 @@ bool AArch64InstrInfo::isCandidateToMergeOrPair(const MachineInstr &MI) const {
       return false;
   }
 
+  // Pairing SVE fills/spills is only valid for little-endian targets that
+  // implement VLS 128.
+  switch (MI.getOpcode()) {
+  default:
+    break;
+  case AArch64::LDR_ZXI:
+  case AArch64::STR_ZXI:
+    return Subtarget.isLittleEndian() &&
+           Subtarget.getSVEVectorSizeInBits() == 128;
```

Comment:
```suggestion
    if (!Subtarget.isLittleEndian() ||
        Subtarget.getSVEVectorSizeInBits() != 128)
      return false;
```
So that we continue to bail based on the later check plus if anything new is added.

Reply:
Thanks, sorted.